sdks/python/apache_beam/metrics/metric.py

```diff
@@ -370,7 +370,27 @@ def get_fq_name(
 
   def add(
       self, system: str, *segments: str, subtype: Optional[str] = None) -> None:
-    self.metric.add(self.get_fq_name(system, *segments, subtype=subtype))
+    """
+    Adds the given details as Lineage.
+
+    For asset level lineage the resource location should be specified as
+    Dataplex FQN, see
+    https://cloud.google.com/data-catalog/docs/fully-qualified-names
+
+    Example of adding FQN components:
+
+    - `add("system", "segment1", "segment2")`
+    - `add("system", "segment1", "segment2", subtype="subtype")`
+
+    Example of adding a FQN:
+
+    - `add("system:segment1.segment2")`
+    - `add("system:subtype:segment1.segment2")`
+    """
+    if len(segments) == 0 and subtype is None:
+      self.metric.add(system)
```

Comment:
strictly speaking this is "system_or_fqn" however naming change may break positional parameters so the original name was kept.

Reply:
It appears we are overloading the system parameter which can be confusing can we add a separate optional parameter for fqn string say "details"

details=None

    if details:
      self.metric.append(details)  
    elif system and segments:
      self.metric.append(self.get_fq_name(system, subtype, segments))  # Add the constructed FQN

Reply:
then one has to call `add(details=...)` to access to this branch which is less preferable either. Let me assign it into an intermediate variable (`system_or_fqn = system`)